espressif/vscode-esp-idf-extension
# Add optional parameter for tagging in the log file

## Description

Add tag labels for existing info, error and warning messages.

Fixes # ([VSC-882](https://jira.espressif.com:8443/browse/VSC-882))

## Type of change

- New feature (non-breaking change which adds functionality)

## How has this been tested?

1. Launch VSCode

2. Run any Espressif project and test out features.

3. All the logs that get gathered in esp_idf_vsc_ext.log should have a "tags" property. The tags should be displayed in the "message" property as well, as can be seen in the example bellow:
Ex: 
`{"tags":["ESP-IDF Utils","execChildProcess"],"message":"[ESP-IDF Utils][execChildProcess]Open On-Chip Debugger  v0.11.0-esp32-20220411 (2022-04-11-08:47)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n","stack":"Error: Open On-Chip Debugger  v0.11.0-esp32-20220411 (2022-04-11-08:47)\nLicensed under GNU GPL v2\nFor bug reports, read\n\thttp://openocd.org/doc/doxygen/bugs.html\n\n\tat /Users/radurentea/Documents/GitHub/vscode-esp-idf-extension/dist/extension.js:100405:86\n\tat ChildProcess.exithandler (node:child_process:399:7)\n\tat ChildProcess.emit (node:events:526:28)\n\tat maybeClose (node:internal/child_process:1092:16)\n\tat Socket.<anonymous> (node:internal/child_process:451:11)\n\tat Socket.emit (node:events:526:28)\n\tat Pipe.<anonymous> (node:net:687:12)\n\tat Pipe.callbackTrampoline (node:internal/async_hooks:130:17)","level":"error","timestamp":"2022-10-31T11:48:48.167Z"}
`

**Test Configuration**:
* ESP-IDF Version: 4.4
* OS: macOS

## Checklist
